addons/website/static/src/js/recaptcha.js

```diff
@@ -0,0 +1,59 @@
+odoo.define('reCaptchaV3Mixin', function (require) {
+"use strict";
+
+/**
+ * The ReCaptchaV3Mixin is a mixin, designed to add the google reCaptchaV3
+ */
+return {
+    /**
+     * @override
+     */
+    start: function () {
+        this.publicKey = odoo.reCaptchaPublicKey;
+        if (this.publicKey) {
+            this._addRecaptchaLegal();
+            const url = `https://www.recaptcha.net/recaptcha/api.js?render=${this.publicKey}`;
+            let script = document.head.querySelector(`script[src="${url}"]`);
+            if (!script) {
+                script = document.createElement("script");
+                script.src = url;
+                script.async = false;
+                document.head.appendChild(script);
+            }
+            this.scriptLoaded = new Promise((resolve) => {
+                script.addEventListener('load', () => {
+                    window.grecaptcha.ready(() => resolve());
+                });
+            });
+        }
+    },
```

Comment:
Should be a class instance instead of a mixin (if not a service but maybe overkill as probably only one object per page would need it).

The code here is basically a copy of ajax.loadJS. And I doubt it works with two repatcha on the same page here. You check if the script exists but if it is already loaded when you enter a second recaptact start it will never be considerered loaded for that one. Bref -> use the ajax.loadJS implementation instead of doing your own.

```javascript
const Class = require('web.Class');
const ServiceMixin = require('...');

const Recaptcha = Class.extend(ServiceMixin, {
    init: function () {
        // ... here maybe you don't need ServiceMixin and you can just us ajax.loadJS
        // directly, not sure (I code without searching here)
        this._scriptLoaded = this._loadLibs({jsLibs: [`https://www.recaptcha.net/recaptcha/api.js?render=${this.publicKey}`],})
            .then(() => new Promise(resolve => window.grepcaptcha.ready(() => resolve())));
    },
    getReCaptchaV3Token: // what you did but public
});

// ... then in website form

init: function () {
    this._recaptcha = new Recaptcha(this);
    // ... and you can use the object calling its public methods.
},
```